scipy/stats/_stats_py.py

```diff
@@ -3911,28 +3911,41 @@ def trim_mean(a, proportiontocut, axis=0):
 F_onewayResult = namedtuple('F_onewayResult', ('statistic', 'pvalue'))
 
 
-def _create_f_oneway_nan_result(shape, axis):
+def _create_f_oneway_nan_result(shape, axis, samples):
     """
     This is a helper function for f_oneway for creating the return values
     in certain degenerate conditions.  It creates return values that are
     all nan with the appropriate shape for the given `shape` and `axis`.
     """
     axis = normalize_axis_index(axis, len(shape))
     shp = shape[:axis] + shape[axis+1:]
-    if shp == ():
-        f = np.nan
-        prob = np.nan
-    else:
-        f = np.full(shp, fill_value=np.nan)
-        prob = f.copy()
-    return F_onewayResult(f, prob)
+    f = np.full(shp, fill_value=_get_nan(*samples))
+    prob = f.copy()
+    return F_onewayResult(f[()], prob[()])
 
 
 def _first(arr, axis):
     """Return arr[..., 0:1, ...] where 0:1 is in the `axis` position."""
     return np.take_along_axis(arr, np.array(0, ndmin=arr.ndim), axis)
 
 
+def _f_oneway_is_too_small(samples, kwargs, axis=-1):
+    if any(sample.shape[axis] == 0 for sample in samples):
+        msg = ('all input arrays have length 1.  f_oneway requires that at '
+               'least one input has length greater than 1.')
+        warnings.warn(stats.DegenerateDataWarning(msg), stacklevel=2)
```

Comment:
Notice that the warning messages are present both here and in the function. I wasn't able to pull them out of the function since the decorator doesn't call `too_small` for 1-D inputs. So, there's a bit of duplication here.

Reply:
Usually, the original functions raise errors for too small 1D input, and the decorator returns NaNs for ND input. We could emit a warning similar to the error message to explain the NaNs, but we don't.

Here, the original function happens to emit a warning for too small 1D input. I don't think we're obligated to do so for ND input.

These warning and errors are very inconsistent in stats, and TBH we need to take a pass just for this sort of thing (as we do for dtype support). So I wouldn't let it complicate things in this PR.

Reply:
I see the thumbs up. Does this mean you will remove the warning for >1D arrays from this function?

Reply:
There are a few failing tests when I remove it. I am thinking of keeping it to preserve backward compatibility. Is there another fix for it?

Reply:
Oops I didn't notice that `f_oneway` already supported `axis`. I guess this is OK. Could you call this function inside the function code to reduce duplication?

Reply:
> Could you call this function inside the function code to reduce duplication?

Will do!